odoo/models.py

```diff
@@ -1637,7 +1637,17 @@ def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get
         # optimize out the default criterion of ``like ''`` that matches everything
         elif not (name == '' and operator in ('like', 'ilike')):
             aggregator = expression.AND if operator in expression.NEGATIVE_TERM_OPERATORS else expression.OR
-            domain = aggregator([[(field_name, operator, name)] for field_name in search_fnames])
+            domains = []
+            for field_name in search_fnames:
+                root_field_name = field_name.split('.')[0]
+                if self._fields[root_field_name].relational or operator == "in":
```

Comment:
This also makes no sense for like-operators which should have the value unchanged.
But the `operator == 'in'` is incorrect. You need to apply this to each value in the 'in'/'not in' operators.

Reply:
I have no idea what this is.
My original fixup was https://github.com/odoo/odoo/commit/104c5b73616af574af7b127d70ed9bab06dcccab

And I think it was working fine